tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -0,0 +1,767 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <zephyr/fff.h>
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/bap.h>
+#include <zephyr/bluetooth/audio/pacs.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/uuid.h>
+
+#include "host_mocks/assert.h"
+#include "mocks/ascs_help_utils.h"
+#include "mocks/bap_unicast_server.h"
+#include "mocks/conn.h"
+#include "mocks/gatt.h"
+#include "mocks/gatt_expects.h"
+#include "mocks/gatt_help_utils.h"
+#include "mocks/iso.h"
+#include "mocks/kernel.h"
+#include "mocks/pacs.h"
+
+#include "ascs_internal.h"
+
+DEFINE_FFF_GLOBALS;
+
+static const struct bt_gatt_attr *ascs_get_attr(const struct bt_uuid *uuid, uint16_t n)
+{
+	const struct bt_gatt_service_static *ascs = ascs_get_gatt_service();
+
+	return gatt_attr_lookup_uuid(ascs, uuid, n);
+}
+
+static void fff_reset_rule_before(const struct ztest_unit_test *test,
+				  void *fixture)
+{
+	BAP_UNICAST_SERVER_FFF_FAKES_LIST(RESET_FAKE);
+	CONN_FFF_FAKES_LIST(RESET_FAKE);
+	GATT_FFF_FAKES_LIST(RESET_FAKE);
+	ISO_FFF_FAKES_LIST(RESET_FAKE);
+	KERNEL_FFF_FAKES_LIST(RESET_FAKE);
+	PACS_FFF_FAKES_LIST(RESET_FAKE);
+}
+
+ZTEST_RULE(fff_reset_rule, fff_reset_rule_before, NULL);
+
+ZTEST_SUITE(ascs_attrs_test_suite, NULL, NULL, NULL, NULL, NULL);
+
+ZTEST(ascs_attrs_test_suite, test_has_sink_ase_chrc)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);
+
+	zassert_not_null(ascs_get_attr(BT_UUID_ASCS_ASE_SNK, 1));
+	zassert_not_null(ascs_get_attr(BT_UUID_ASCS_ASE_SNK, CONFIG_BT_ASCS_ASE_SNK_COUNT));
+}
```

Comment:
if `CONFIG_BT_ASCS_ASE_SNK_COUNT` is 0 then `CONFIG_BT_ASCS_ASE_SNK` is not set, meaning this test will be skipped, because it's guarded with `Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);`